File: backend/api/src/main/scala/com/github/ai/split/api/ErrorMessageDto.scala
package com.github.ai.split.api

import zio.json.{DeriveJsonEncoder, JsonEncoder}

case class ErrorMessageDto(
  message: Option[String],
  exception: String,
  stacktraceBase64: String,
  stacktraceLines: List[String]
)

object ErrorMessageDto {
  implicit val encoder: JsonEncoder[ErrorMessageDto] = DeriveJsonEncoder.gen[ErrorMessageDto]
}

File: backend/api/src/main/java/com/github/ai/split/api/ErrorMessageDto.java
package com.github.ai.split.api;

import java.util.Collections;
import java.util.List;

public class ErrorMessageDto {

    public String message = null;
    public String exception = "";
    public String stacktraceBase64 = "";
    public List<String> stacktraceLines = Collections.emptyList();

    public ErrorMessageDto() {
    }

    public ErrorMessageDto(String message, String exception, String stacktraceBase64, List<String> stacktraceLines) {
        this.message = message;
        this.exception = exception;
        this.stacktraceBase64 = stacktraceBase64;
        this.stacktraceLines = stacktraceLines;
    }
}
